File: DC_CS422_Lab5/Week5Lab-iOS/Week5Lab-iOS/Week5Lab/entities/Flashcard.swift
//
//  Flashcard.swift
//
//  Created by Andrew Taylor on 1/22/23.
//

import Foundation

class Flashcard: Equatable {
    static func == (lhs: Flashcard, rhs: Flashcard) -> Bool {
        lhs.term == rhs.term && lhs.definition == rhs.definition
    }
    
    var term: String
    var definition: String
    
    init(term: String, definition: String) {
        self.term = term
        self.definition = definition
    }
}

func getFlashcards() -> [Flashcard]
{
    var flashcards = [Flashcard]()
    for i in 1...10
    {
        flashcards.append(Flashcard(term: "Term \(i)", definition: "Definition \(i)"))
    }
    return flashcards
}

File: DC_CS422_Lab4/Week4Lab-Android/app/src/main/java/cs/mad/week4lab/entities/Flashcard.kt
package cs.mad.week4lab.entities

data class Flashcard(var term: String, var definition: String)

fun getFlashcards(): List<Flashcard> {
    val flashcards = mutableListOf<Flashcard>()
    for (i in 0..9) {
        flashcards.add(Flashcard("Term $i", "Definition $i"))
    }
    return flashcards
}